### PR TITLE
Log errors to the last used channel

### LINKED ISSUE #10: Exceptions should log to the last used channel
# Description
When an exception occurs, the bot should try message the last used channel with the contents of the exception. This will make debugging easier when the bot is live

### PR DESCRIPTION
**Pull request by geniusjam**
* Use promises throughout the code, so that we can minimize the amount of unhandled rejections and shift the burden of handling errors to the parent function
* Maintain a variable `client.lastUsedChannel` and send errors to that channel when `unhandledRejection` and `uncaughtException` events are fired, resolving #10
Resolves #10

### CODE REVIEW

File: bot.js [hunk 1]
```diff
@@ -51,12 +51,26 @@ client.on("disconnect", function() {
...
 // Create an event listener for messages
-client.on('messageCreate', message => {
+client.on('messageCreate', async message => {
     if (client.user.id != message.author.id) {
-        ChatHandler.handle(message.content, message.author, message.channel, message);
+        await ChatHandler.handle(message.content, message.author, message.channel, message);
     }
 });
 
+process.on('uncaughtException', handleException);
```
Comment (StrangeOne101) [inline bot.js:55]: This is good, but you might want to remove the initial uncaughtException handling at the top of the bot file.

**Review by StrangeOne101 — CHANGES_REQUESTED** [review]
Looks good for the most part. Has all the async changes been tested? There's just a lot changed here, so want to verify it's been tested before merging.

It'd also be good if you could get rid of the accidental dupe mentioned, too

File: bot.js (hunk 1 above)
Comment (geniusjam) [reply bot.js:55]: Yeah, probably have missed that

**Comment by geniusjam** [comment]
Yes, they're tested. It's mostly just putting async and await before functions. I have also tested registering regex commands from pastebin links.

**Review by StrangeOne101 — APPROVED** [review]
Sweet

**Pull request merged**